src/Java.Interop.Dynamic/Java.Interop.Dynamic/JavaMethodBase.cs

```diff
@@ -121,7 +121,7 @@ public bool CompatibleWith (List<JniType> args, DynamicMetaObject[] dargs)
 					if (JniEnvironment.Types.GetJniTypeNameFromClass (arguments [i]) != vm.TypeManager.GetTypeSignature (dargs [i].LimitType).Name)
 						return false;
 				}
-				else if (!JniEnvironment.Types.IsAssignableFrom (arguments [i], args [i].PeerReference))
+				else if (!JniEnvironment.Types.IsAssignableFrom (args [i].PeerReference, arguments [i]))
```

Comment:
I needed to swap these items around as it appears that the first item needs to have the second item as the interface: https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/functions.html#wp16061

> Returns JNI_TRUE if either of the following is true:
>
> - The first and second class arguments refer to the same Java class.
> - The first class is a subclass of the second class.
> - The first class has the second class as one of its interfaces.